packages/trader/src/App/Components/Elements/ContractDrawer/__tests__/swipeable-components.spec.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { SwipeableContractAudit, SwipeableContractDrawer } from '../swipeable-components';
+
+const mocked_children = 'Mocked children';
+
+describe('<SwipeableContractAudit />', () => {
+    const default_mocked_props = {
+        is_multiplier: true,
+        onSwipedDown: jest.fn(),
+    };
+
+    const modal = document.createElement('div');
+    beforeAll(() => {
+        document.body.appendChild(modal);
+    });
+    afterAll(() => {
+        document.body.removeChild(modal);
+    });
+
+    it('should render passed children', () => {
+        modal.setAttribute('id', 'dt_contract_drawer_audit');
+        render(
+            <SwipeableContractAudit {...default_mocked_props}>
+                <div>{mocked_children}</div>
+            </SwipeableContractAudit>
+        );
+
+        expect(screen.getByText(mocked_children)).toBeInTheDocument();
+    });
+    it('should not render anything if id in modal is different from dt_contract_drawer_audit', () => {
+        modal.setAttribute('id', 'modal_root');
+        const { container } = render(
+            <SwipeableContractAudit {...default_mocked_props}>
+                <div>{mocked_children}</div>
+            </SwipeableContractAudit>
+        );
+
+        expect(container).toBeEmptyDOMElement();
+    });
+});
+
+describe('<SwipeableContractDrawer />', () => {
+    const default_mocked_props = {
+        onSwipedDown: jest.fn(),
```

Comment:
Can we also add test cases to check if `onSwipedDown` and `onSwipedUp` are called?

Reply:
These functions are passed to the 3d party library (`react-swipeable`) and called from there via ref. I tried all possible userevent and fireevent methods but these callbacks don't get called. Could you please give an example if it can be tested?

Reply:
<img width="356" alt="image" src="https://github.com/binary-com/deriv-app/assets/56330681/1f169683-10e5-4258-b629-942793de1e0b">
Looks like they simulate using multiple touch events - https://github.com/FormidableLabs/react-swipeable/blob/main/__tests__/useSwipeable.spec.tsx

Yeah, since the unit testing is covered in the library itself, we don't have to do it. Thanks for checking 🙏